purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/OfflineEntitlementsIntegrationTest.kt

```diff
@@ -0,0 +1,394 @@
+package com.revenuecat.purchases
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.common.sha1
+import com.revenuecat.purchases.factories.StoreProductFactory
+import com.revenuecat.purchases.factories.StoreTransactionFactory
+import com.revenuecat.purchases.helpers.mockQueryProductDetails
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import com.revenuecat.purchases.models.StoreTransaction
+import io.mockk.every
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.After
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.time.Duration
+
+@Suppress("TooManyFunctions")
+@RunWith(AndroidJUnit4::class)
+class OfflineEntitlementsIntegrationTest : BasePurchasesIntegrationTest() {
```

Comment:
Note that I didn't add automated tests that tested offline entitlements with multiple purchases. Reasons being:
- I would need to assume many things about how Google works in order to mock it realistically. I really wish Google released something to test purchases more easily :(.
- We would need another valid token for a different product if we wanted to test it against the backend